docs/design/mono/web/coop-suspend.md

```diff
@@ -0,0 +1,245 @@
+---
+title: Runtime Cooperative Suspend
+---
```

Comment:
Some of this is also covered in 
 https://github.com/dotnet/runtime/blob/main/docs/design/mono/mono-thread-state-machine.md

but some of the info is only here (for example: `MONO_REQ_GC_UNSAFE` etc)